src/compiler/program.ts

```diff
@@ -2699,11 +2699,12 @@ namespace ts {
                     projectReferenceRedirects.set(sourceFilePath, false);
                     return undefined;
                 }
-                sourceFile.path = sourceFilePath;
-                sourceFile.resolvedPath = sourceFilePath;
-                sourceFile.originalFileName = refPath;
                 commandLine = parseJsonSourceFileConfigFileContent(sourceFile, configParsingHost, basePath, /*existingOptions*/ undefined, refPath);
             }
+            sourceFile.path = sourceFilePath;

```

Comment:
I think the fix is correct but we should assert sourceFile.path === undefined || sourceFile.path == sourceFilePath so as to make sure we aren't overriding paths incorrectly especially since we don't get sourceFile explicitly.